### PR TITLE
WIP: OpenAPI spec linter

### PR DESCRIPTION
**Pull request by lakshmi-kannan**
## Output

```
(virtualenv)vagrant@st2dev /m/s/s/st2 ❯❯❯ make .lint-api-spec                                            feature/lint_openapi_spec ✭ ✱ ◼

================== Lint API spec ====================

st2common/bin/st2-validate-api-spec --generate
2017-05-03 16:36:20,105 ERROR [-] API model field "x-api-model" not defined for definition "TokenValidationRequest".
2017-05-03 16:36:20,107 ERROR [-] API model field "x-api-model" not defined for definition "PolicyCreate".
2017-05-03 16:36:20,107 ERROR [-] API model field "x-api-model" not defined for definition "PacksRegister".
2017-05-03 16:36:20,108 ERROR [-] API model field "x-api-model" not defined for definition "TriggerRequest".
2017-05-03 16:36:20,109 ERROR [-] API model field "x-api-model" not defined for definition "PolicyTypeList".
2017-05-03 16:36:20,110 ERROR [-] API model field "x-api-model" not defined for definition "SensorType".
2017-05-03 16:36:20,110 ERROR [-] API model field "x-api-model" not defined for definition "Trigger".
2017-05-03 16:36:20,111 ERROR [-] API model field "x-api-model" not defined for definition "KeyValuePair".
2017-05-03 16:36:20,111 ERROR [-] API model field "x-api-model" not defined for definition "PacksContentRegisterType".
2017-05-03 16:36:20,114 ERROR [-] API model field "x-api-model" not defined for definition "TracesList".
2017-05-03 16:36:20,114 ERROR [-] API model field "x-api-model" not defined for definition "TriggerTypeRequest".
2017-05-03 16:36:20,115 ERROR [-] API model field "x-api-model" not defined for definition "PacksList".
2017-05-03 16:36:20,116 ERROR [-] API model field "x-api-model" not defined for definition "AliasExecution".
2017-05-03 16:36:20,116 ERROR [-] API model field "x-api-model" not defined for definition "RuleType".
2017-05-03 16:36:20,117 ERROR [-] API model field "x-api-model" not defined for definition "PacksInstall".
2017-05-03 16:36:20,119 ERROR [-] API model field "x-api-model" not defined for definition "RunnerType".
2017-05-03 16:36:20,120 ERROR [-] API model field "x-api-model" not defined for definition "ActionAliasMatchRequest".
2017-05-03 16:36:20,121 ERROR [-] API model field "x-api-model" not defined for definition "ActionParametersSubSchema".
2017-05-03 16:36:20,121 ERROR [-] API model field "x-api-model" not defined for definition "ActionAliasRequest".
2017-05-03 16:36:20,121 ERROR [-] API model field "x-api-model" not defined for definition "Webhook".
2017-05-03 16:36:20,121 ERROR [-] API model field "x-api-model" not defined for definition "PacksUninstall".
2017-05-03 16:36:20,121 ERROR [-] API model field "x-api-model" not defined for definition "WebhookBody".
2017-05-03 16:36:20,121 ERROR [-] API model field "x-api-model" not defined for definition "TriggerInstance".
2017-05-03 16:36:20,121 ERROR [-] API model field "x-api-model" not defined for definition "TokenValidationResult".
2017-05-03 16:36:20,121 ERROR [-] API model field "x-api-model" not defined for definition "LiveAction".
2017-05-03 16:36:20,121 ERROR [-] API model field "x-api-model" not defined for definition "PackConfig".
2017-05-03 16:36:20,122 ERROR [-] API model field "x-api-model" not defined for definition "ActionAliasHelp".
2017-05-03 16:36:20,124 ERROR [-] API model field "x-api-model" not defined for definition "PacksSearchShow".
2017-05-03 16:36:20,124 ERROR [-] API model field "x-api-model" not defined for definition "NotificationPropertySubSchema".
2017-05-03 16:36:20,125 ERROR [-] API model field "x-api-model" not defined for definition "RuleEnforcementsList".
2017-05-03 16:36:20,126 ERROR [-] API model field "x-api-model" not defined for definition "PackIndex".
2017-05-03 16:36:20,126 ERROR [-] API model field "x-api-model" not defined for definition "ActionAliasMatch".
2017-05-03 16:36:20,126 ERROR [-] API model field "x-api-model" not defined for definition "ValidationError".
2017-05-03 16:36:20,127 ERROR [-] API model field "x-api-model" not defined for definition "TimersList".
2017-05-03 16:36:20,127 ERROR [-] API model field "x-api-model" not defined for definition "PackView".
2017-05-03 16:36:20,129 ERROR [-] API model field "x-api-model" not defined for definition "Token".
2017-05-03 16:36:20,130 ERROR [-] API model field "x-api-model" not defined for definition "DataFilesSubSchema".
2017-05-03 16:36:20,130 ERROR [-] API model field "x-api-model" not defined for definition "Error".
2017-05-03 16:36:20,130 ERROR [-] API model field "x-api-model" not defined for definition "TokenRequest".
2017-05-03 16:36:20,131 ERROR [-] API model field "x-api-model" not defined for definition "KeyValuePairRequest".
2017-05-03 16:36:20,132 ERROR [-] API model field "x-api-model" not defined for definition "ExecutionFilters".
2017-05-03 16:36:20,132 ERROR [-] API model field "x-api-model" not defined for definition "PolicyList".
2017-05-03 16:36:20,132 ERROR [-] API model field "x-api-model" not defined for definition "Execution".
2017-05-03 16:36:20,133 ERROR [-] API model field "x-api-model" not defined for definition "PackConfigCreate".
2017-05-03 16:36:20,134 ERROR [-] API model field "x-api-model" not defined for definition "ActionParameters".
2017-05-03 16:36:20,134 ERROR [-] API model field "x-api-model" not defined for definition "TriggerType".
2017-05-03 16:36:20,134 ERROR [-] API model field "x-api-model" not defined for definition "ExecutionRequest".
2017-05-03 16:36:20,134 ERROR [-] API model field "x-api-model" not defined for definition "PackConfigView".
2017-05-03 16:36:20,134 ERROR [-] API model field "x-api-model" not defined for definition "PackConfigContent".
make: *** [.lint-api-spec] Error 1
(virtualenv)vagrant@st2dev /m/s/s/st2 ❯❯❯
```

### CODE REVIEW

File: st2common/st2common/openapi.yaml [hunk 1]
```diff
@@ -3750,15 +3750,15 @@ definitions:
       status:
         description: The current status of the action execution.
         type: string
-        enum: {{ LIVEACTION_STATUSES }}
```
Comment (Kami) [inline st2common/st2common/openapi.yaml]: To confirm - this file is automatically generated based on .j2 definition on make target and build?

**Comment by Kami** [comment]
So far LGTM :+1: 

Can you please gist the sample output?

File: st2common/st2common/openapi.yaml (hunk 1 above)
Comment (lakshmi-kannan) [inline st2common/st2common/openapi.yaml]: I haven't hooked up a make target yet. I plan to do that.

**Comment by Kami** [comment]
LGTM for now once the tests pass.

Next step will be fixing / addressing all the errors :P

**Comment by lakshmi-kannan** [comment]
Closing. Opened  https://github.com/StackStorm/st2/pull/3462 instead.

**Pull request closed without merging**